internal/transport/http2_client.go

```diff
@@ -1266,23 +1267,28 @@ func (t *http2Client) operateHeaders(frame *http2.MetaHeadersFrame) {
 		// that the peer is speaking gRPC and we are in gRPC mode.
 		isGRPC         = !initialHeader
 		mdata          = make(map[string][]string)
-		contentTypeErr string
+		contentTypeErr = "malformed header: missing HTTP content-type"
 		grpcMessage    string
 		statusGen      *status.Status
-
-		httpStatus string
-		rawStatus  string
+		httpStatusCode *int
+		httpStatusErr  string
+		rawStatus      string
 		// headerError is set if an error is encountered while parsing the headers
 		headerError string
 	)
 
+	if !endStream {
```

Comment:
Actually I believe this is incorrect, as it will not validate that a trailers-only response contains the HTTP status.  Changing this to `if initialHeader` should fix this.

Reply:
Ping on this since it's still something we should fix before merging.

Reply:
Fixed! Sorry for the delay.